ilastik/applets/pixelClassification/pixelClassificationGui.py

```diff
@@ -718,46 +702,42 @@ def setPredLayerName(n, predictLayer_=predictLayer, initializing=False):
-        logger.debug("toggling interactive mode to '%r'" % checked)
+    def hasFeatures(self) -> bool:
+        feature_images_slot = self.topLevelOperatorView.FeatureImages
+        return feature_images_slot.ready() and feature_images_slot.meta.shape != None
 
-        if checked==True:
-            if not self.topLevelOperatorView.FeatureImages.ready() \
-            or self.topLevelOperatorView.FeatureImages.meta.shape==None:
-                self.labelingDrawerUi.liveUpdateButton.setChecked(False)
-                self.labelingDrawerUi.liveUpdateButton.setIcon(QIcon(ilastikIcons.Play))
-                self.labelingDrawerUi.suggestFeaturesButton.setEnabled(False)
-                mexBox=QMessageBox()
-                mexBox.setText("There are no features selected ")
-                mexBox.exec_()
-                return
+    def getLiveUpdateFlag(self):
+        return self.labelingDrawerUi.liveUpdateButton.isChecked()
 
-        # If we're changing modes, enable/disable our controls and other applets accordingly
-        if self.interactiveModeActive != checked:
-            self.interactiveModeActive = checked
-            if checked:
-                self.labelingDrawerUi.liveUpdateButton.setIcon(QIcon(ilastikIcons.Pause))
-                self.labelingDrawerUi.labelListView.allowDelete = False
-                self.labelingDrawerUi.AddLabelButton.setEnabled( False )
-            else:
-                self.labelingDrawerUi.liveUpdateButton.setIcon(QIcon(ilastikIcons.Play))
-                num_label_classes = self._labelControlUi.labelListModel.rowCount()
-                self.labelingDrawerUi.labelListView.allowDelete = ( num_label_classes > self.minLabelNumber )
-                self.labelingDrawerUi.AddLabelButton.setEnabled( ( num_label_classes < self.maxLabelNumber ) )
+    def setLiveUpdateFlag(self, checked:bool):
+        liveUpdateButton = self.labelingDrawerUi.liveUpdateButton
+        liveUpdateButton.blockSignals(True)
+        liveUpdateButton.setChecked(checked)
+        liveUpdateButton.blockSignals(False)
 
-            self.labelingDrawerUi.suggestFeaturesButton.setEnabled(not checked)
-            self.labelingDrawerUi.liveUpdateButton.setChecked(checked)
+        self.topLevelOperatorView.FreezePredictions.setValue(not checked)
 
-        self.topLevelOperatorView.FreezePredictions.setValue( not checked )
+        if checked and self.hasFeatures():
+            self.labelingDrawerUi.labelListView.allowDelete = False
+            self.labelingDrawerUi.AddLabelButton.setEnabled( False )
 
-        # Auto-set the "show predictions" state according to what the user just clicked.
-        if checked:
             self._viewerControlUi.checkShowPredictions.setChecked( True )
             self.handleShowPredictionsClicked()
+        else:
+            self.labelingDrawerUi.suggestFeaturesButton.setEnabled(False)
+            if not self.hasFeatures():
+                mexBox=QMessageBox()
```

Comment:
```suggestion
                mexBox = QMessageBox()
```

Or maybe even something like `QMessageBox.warning(self, "Features", "No features selected")`.

Also, maybe remove nested condition by duplicating the `suggestFeaturesButton.setEnabled(False)` call? Seems a bit easier to follow.

```python
if checked and self.hasFeatures():
    ...
elif not self.hasFeatures():
    self.labelingDrawerUi.suggestFeaturesButton.setEnabled(False)
    QMessageBox.warning(self, "Features", "No features selected")
    return
else:
    self.labelingDrawerUi.suggestFeaturesButton.setEnabled(False)
    ....
```

Reply:
Yeah, that part did feel weird. Turns out it was needlessly convoluted. I've streamlined the code a bit more